catch-uncommitted.sh

```diff
@@ -39,7 +39,7 @@ if [ $catch_no_git -eq 1 ] && !(hash git 2> /dev/null); then
 elif ! git diff HEAD --exit-code ${ignore_eol} -- . "${extra_args[@]}" ; then
 	echo '** Uncommitted changes found (^^^ diff above ^^^) - FAIL **'
 	exit 1
-elif test -n "$(git ls-files --exclude-standard --others | tee /dev/tty)" ; then
+elif test -n "$(git ls-files --exclude-standard --others)" ; then
 	echo '** Untracked uncommitted files found (^^^ listed above ^^^) - FAIL **'
```

Comment:
I think (didn't test) that a consequence of removing `| tee /dev/tty` is that the output is no longer printed to the console, which means that the `echo` message `"^^^ listed above ^^^"` would not match the output and it would be harder to debug issues when `catch-uncommitted` failed at that point. There may be another way of printing the message to the console without using `/dev/tty`, perhaps something like:
```bash
files = "$(git ls-files --exclude-standard --others)"
if test -n files ; then
    echo '** FAIL - Untracked uncommitted files found: **'
    echo "$files"
```
But the above needs adjusting to fit the full if/then/else chain.

Reply:
@pdcastro This was spot on, thanks for the advice. Tested to work on my local machine and during `docker build`.